pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -324,6 +324,10 @@ impl<T: Config> Module<T> {
             }
         };
 
+        // Cleanup contract billing information map
+        ContractBillingInformationByID::remove(contract_id);
```

Comment:
Won't this cause the last consumption report to not be billed anymore?

Reply:
Yes it would, hmm. Maybe we can do a check to see if has 0 unbilled, if so, remove it?